core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/OracleExportAction.java

```diff
@@ -0,0 +1,235 @@
+    switch (format) {
+      case "csv":
+        return ",";
+      case "tsv":
+        return "  ";
+      case "psv":
+        return "|";
+      default:
+        throw new IllegalArgumentException(
+          String.format("Invalid format '%s'. Must be one of %s", format, Joiner.on(',').join(FORMATS)));
+    }
+  }
+
+  /**
+   * Config class that contains all properties necessary to execute the SQLPLUs spool command.
+   */
+  public static class OracleExportActionConfig extends PluginConfig {
+
+    @Description("Host name of the remote DB machine")
+    @Macro
+    private String oracleServerHostname;
+
+    @Nullable
+    @Description("Port of the remote DB machine. Defaults to 22")
+    @Macro
+    private Integer oracleServerPort;
+
+    @Description("Username for remote DB host")
+    @Macro
+    private String oracleServerUsername;
+
+    @Description("Password for remote DB host")
+    @Macro
+    private String oracleServerPassword;
+
+    @Description("Username to connect to oracle DB")
+    @Macro
+    private String dbUsername;
+
+    @Description("Password to connect to oracle DB")
+    @Macro
+    private String dbPassword;
+
+    @Description("Path of the ORACLE_HOME")
+    @Macro
+    private String oracleHome;
+
+    @Description("Oracle SID")
+    @Macro
+    private String oracleSID;
+
+    @Description("Query to be executed for export")
+    @Macro
+    private String queryToExecute;
+
+    @Description("Path where output file to be exported")
+    @Macro
+    private String pathToWriteFinalOutput;
+
+    @Description("Format of the output file")
+    @Macro
+    private String format;
+
+    public OracleExportActionConfig() {
+      this.oracleServerPort = 22;
+    }
+
+    public OracleExportActionConfig(String oracleServerHostname, @Nullable Integer oracleServerPort,
+                                    String oracleServerUsername, String oracleServerPassword,
+                                    String dbUsername, String dbPassword,
+                                    String oracleHome, String oracleSID,
+                                    String pathToWriteFinalOutput, String queryToExecute, String format) {
+      this.oracleServerHostname = oracleServerHostname;
+      this.oracleServerPort = oracleServerPort;
+      this.oracleServerUsername = oracleServerUsername;
+      this.oracleServerPassword = oracleServerPassword;
+      this.dbUsername = dbUsername;
+      this.dbPassword = dbPassword;
+      this.oracleHome = oracleHome;
+      this.oracleSID = oracleSID;
+      this.pathToWriteFinalOutput = pathToWriteFinalOutput;
+      this.queryToExecute = queryToExecute;
+      this.format = format;
+    }
+
+    public void validate() {
+      if (!containsMacro("oracleServerPort") && oracleServerPort < 0) {
+        throw new IllegalArgumentException("Port cannot be negative");
+      }
+      if (!containsMacro(queryToExecute) && !queryToExecute.endsWith(";")) {
```

Comment:
shouldn't have this check for semi-colon like in my earlier comment. Should instead correct it if it does not end in a semi colon